oshmem/mca/spml/ucx/spml_ucx.c

```diff
@@ -267,7 +409,11 @@ int mca_spml_ucx_ctx_mkey_del(mca_spml_ucx_ctx_t *ucx_ctx, int pe, uint32_t segn
     ucp_peer_t *ucp_peer;
     int rc;
     ucp_peer = &(ucx_ctx->ucp_peers[pe]);
-    ucp_rkey_destroy(ucx_mkey->rkey);
+    if (!oshmem_proc_on_local_node(pe)) {
```

Comment:
why need? we won't find it anyway..

Reply:
I think it is needed because when we don't find it, we insert it and consume entries that we know won't be later matched. (with ppn 256, it means 2 segs * 256 = 512 entries which could be a lot if we only use rkey store count of 1000)

Reply:
where would we insert it if not found? isn't it the cleanup flow?

Reply:
ah right, this is release path, yes we can omit